src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php

```diff
@@ -22,108 +23,161 @@ class AblyBroadcaster extends Broadcaster
      */
     protected $ably;
 
+    /**
+     * Used for setting expiry of issues tokens.
+     *
+     * @var int|mixed
+     * @default 1 hr
+     */
+    private $tokenExpiry = 3600;
+
+    /**
+     * Default channel capabilities, all public channels are by default given subscribe, history and channel-metadata access
+     * Set as per https://ably.com/docs/core-features/authentication#capability-operations.
+     *
+     * @var array|\string[][]
+     */
+    private $defaultChannelClaims = [
+        'public:*' => ['subscribe', 'history', 'channel-metadata'],
+    ];
+
     /**
      * Create a new broadcaster instance.
      *
-     * @param  \Ably\AblyRest  $ably
+     * @param $ably \Ably\AblyRest
      * @return void
      */
-    public function __construct(AblyRest $ably)
+    public function __construct(AblyRest $ably, $config)
     {
         $this->ably = $ably;
+        if (self::$serverTimeDiff == null) {
+            self::setServerTime(round($this->ably->time() / 1000));
+        }
+        if (array_key_exists('disable_public_channels', $config) && $config['disable_public_channels']) {
+            $this->defaultChannelClaims = ['public:*' => ['channel-metadata']];
+        }
+        if (array_key_exists('token_expiry', $config)) {
+            $this->tokenExpiry = $config['token_expiry'];
+        }
     }
 
+    private static $serverTimeDiff = null;
+
     /**
-     * Authenticate the incoming request for a given channel.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return mixed
-     *
-     * @throws \Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException
+     * @param $time int
```

Comment:
```suggestion
     * @param  int  $time
```
First variable type and then the variable name

Reply:
Fixed!